public/app/features/search/hooks/useShowDashboardPreviews.ts

```diff
@@ -0,0 +1,14 @@
+import { PREVIEWS_LOCAL_STORAGE_KEY } from '../constants';
+import { config, reportInteraction } from '@grafana/runtime/src';
+import { useLocalStorage } from 'react-use';
+
+export const useShowDashboardPreviews = () => {
+  const previewsFeatureEnabled = Boolean(config.featureToggles.dashboardPreviews);
+  const [showPreviews, setShowPreviews] = useLocalStorage<boolean>(PREVIEWS_LOCAL_STORAGE_KEY, previewsFeatureEnabled);
+  const onShowPreviewsChange = (showPreviews: boolean) => {
+    reportInteraction(`${showPreviews ? 'enabled' : 'disabled'}_dashboard_previews`);
```

Comment:
So this will report the one time it gets toggled?  The value is saved in local storage... So if turned on it stays on.

Maybe we should report the state, not the action?

Reply:
Are you suggesting to just report the state every time the component rerenders? I think reporting the action should be enough for all the metrics we want, assuming the data passed to `reportInteraction` ends up in rudderstack at least _most of the time_. We will be able to infer % of customers that... 1. use the feature (based on feature rollout %), 2. keep it on, 3. flip between off/on.